src/firejail/profile.c

```diff
@@ -113,12 +113,85 @@ void profile_add_ignore(const char *str) {
 }
 
 
+int profile_check_conditional(char *ptr, int lineno, const char *fname) {
+	struct cond_t {
+		char *name;	// conditional name
+		size_t len;	// length of name
+		bool value;	// true if set
+	} conditionals[] = {
+		{"HAS_APPIMAGE", strlen("HAS_APPIMAGE"), arg_appimage!=0},
+		NULL
+	}, *cond = conditionals;
+	char *tmp = ptr;
+
+	if (*ptr++ != '?')
+		return 1;
+
+	while (cond->name) {
+		// continue if not this conditional
+		if (strncmp(ptr, cond->name, cond->len) != 0) {
+			cond++;
+			continue;
+		}
+
+		// if set, continue processing statement in caller
+		if (cond->value) {
+			// move ptr to start of profile line
+			ptr = strdup(ptr + cond->len + 1);
```

Comment:
I think you are out-of-bounds of the old ptr if nothing follows after the conditional

Reply:
Good catch!  This is a perfect example of why I want as much code as possible not running `setuid`...